torch/onnx/_internal/registration.py

```diff
@@ -303,5 +312,69 @@ def _register_module(module, opset: OpsetVersion) -> None:
             registry.register(f"aten::{name}", opset, obj)
 
 
+@_beartype.beartype
+def onnx_symbolic(
+    name: str,
+    opset: Union[OpsetVersion, Sequence[OpsetVersion]],
+    decorate: Optional[Sequence[Callable]] = None,
+    custom: bool = False,
+) -> Callable:
+    """Registers a symbolic function.
+
+    Usage::
+
+    ```
+    @onnx_symbolic("aten::symbolic_b", opset=10, decorate=[quantized_aten_handler(scale=1/128, zero_point=0)])
+    @symbolic_helper.parse_args("v", "v", "b")
+    def symbolic_b(g: _C.Graph, x: _C.Value, y: _C.Value, arg1: bool) -> _C.Value:
+        ...
+    ```
+
+    Args:
+        name: The qualified name of the function.
+        opset: The opset versions of the function to register at.
+        decorate: A sequence of decorators to apply to the function.
+        custom: Whether the function is a custom symbolic function.
+    """
+
+    def wrapper(func: Callable) -> Callable:
+        decorated = func
+        if decorate is not None:
+            for decorate_func in decorate:
+                decorated = decorate_func(decorated)
+
+        global registry
+        nonlocal opset
+        if isinstance(opset, OpsetVersion):
+            opset = (opset,)
+        for opset_version in opset:
```

Comment:
I think this contradicts a bit of not repeat registering same function for multiple versions established in the first PR.
Do you envision supporting both kind of usages? I think one selling point for single registration multiple dispatch versions (the existing approach) is relieving the need to update whenever new onnx opset is introduced.

Reply:
This was there to provide a finer grained control for users who wants to register multiple custom versions. That said I don't foresee us using it internally. Do you think I should create a separate implementation for the custom decorator?

Reply:
I think the `version` still refers to onnx version. So that will go through the same process, hence I think it could be a bit confusing when both styles are available.
In fact you bring out a good point, I'm not sure if existing api allows users specifying their custom domain versions. This could be something to add in the custom_onnx_symbolic decorator.

Reply:
It does with a strange behavior, where if you register at version `n`, all `n..MAX_VERSION` versions are overridden. In ort we typically see users register at version 1, effectively overriding all versions.

Reply:
sg. This is fine if it was the old behavior anyways. 
In my comment above I was thinking about custom domain version. it appears in export api we do have `custom_opsets` argument. But I think the exporter do not allow dispatching for custom domain, and the custom domain ops in custom symbolic function are expected to match the version in `custom_opsets`. I think this can be greatly improved with the custom op decorator.

Reply:
Is there any changes you suggested we make on the custom op decorator?